### ISSUE TITLE
SST omega is wrong in sharp corners

### ISSUE THREAD

**Issue by bigfooted** [su2code/SU2#2383]
**Describe the bug**

In a 3D channel with square cross section and inflation layers, omega can be wrong in the corner. Below is a cross section of such a channel. Omega is computed using: 
$$\omega_{wall} = 60 \nu / (\beta d^2)$$, 
with d the "distance to the next point away from the wall" [Menter, 1994]. We take the point that is most normal to the wall, computed using FindNormal_Neighbor. However, in the case below, the returned points are the points on the wall. Since they are very far away, the corner omega is very small, order 1e3 instead of 1e6 in the points surrounding the corner point, which causes convergence issues. 

Menter mentions that the interior point should actually have y+ < 3. So we should favor nodes that are closer. Maybe prefer nodes based on a weight factor using the distance and the angle? 

![image](https://github.com/user-attachments/assets/21156570-086b-44ea-bbdd-2381917c8031)

**Comment by bigfooted** [comment]
In any case since the boundary condition is based on an y+ value, I think points on the wall should always be excluded.

**Comment by pcarruscag** [comment]
There are two versions of the w boundary condition, the "60" version at the face/wall and the "6" version on the first wall CV. We can switch to the latter, it's consistent with wall functions.

**Comment by rois1995** [comment]
Is the problem related to the normal of the corner point being computed wrong?

**Comment by pcarruscag** [comment]
Of course, all of this is brought about by the quirks of vertex centered discretizations :)

**Comment by bigfooted** [comment]
@pcarruscag do you mean like the approach of Menter and Esch (What Fluent uses):
https://www.researchgate.net/file.PostFileLoader.html?id=58500561b0366d121d756004&assetKey=AS:438841040478208@1481639265395

This is the paper where they introduce a blending function for the wall bc $\omega(y^+) = (\omega_{visc}(y^+)^2 + \omega_{log}(y^+)^2)^{0.5}$
and they use $\omega_{visc}(y) = \frac{6 \nu}{\beta y^2}$

**Comment by pcarruscag** [comment]
They are both cell centers for us, but one coincides with the wall and the other is the first node of the wall, there are some testing benefits to having a consistent treatment for no-slip and wall-modelled walls. But I don't know if the "60" version at the wall is more robust.

**Comment by rois1995** [comment]
Maybe I am missing something, but in this particular case, if the wall normal on a boundary point is taken as the mean of the boundary edge's normal that are connected to that point, shouldn't the first normal point be the correct one?

**Comment by pcarruscag** [comment]
if the vertical and horizontal walls are different markers there are two vertices that share that node

**Comment by rois1995** [comment]
I see. So the problem only appears if adjacent wall surfaces have different markers. Otherwise the normal should be the averaged one and the first normal point should be correct.

**Comment by bigfooted** [comment]
Points 1 and 3 can be on the same marker for the problem to occur. We do not average. Also, 1 and 3 are on the wall and not in the interior and we need an interior point, which is 2. If we do not allow wall nodes then we will end up with point 2.

**Comment by bigfooted** [comment]
I have a quick fix in the above mentioned PR. What I also noticed is that the wall omega strongly depends on the local mesh. Here is omega on the 90 degree bend:
![image](https://github.com/user-attachments/assets/7c488506-ff9e-4cb8-a457-c503698ef502)

**Comment by pcarruscag** [comment]
This is fixed now, right @bigfooted ?
